contracts/telospooldex/telospooldex.cpp

```diff
@@ -0,0 +1,297 @@
+                funding_attempts.modify(fund_it, from, [&](auto &row) {
+                    row.commodity += quantity;
+                });
+
+            if (quantity.symbol == fund_it->currency.symbol)
+                funding_attempts.modify(fund_it, from, [&](auto &row) {
+                    row.currency += quantity;
+                });
+
+            if (fund_it->commodity.amount == 0 || fund_it->currency.amount == 0)
+                return;
+
+            asset commodity_ex = asset_change_precision(
+                    fund_it->commodity, ARITHMETIC_PRECISION);
+            asset currency_ex = asset_change_precision(
+                    fund_it->currency, ARITHMETIC_PRECISION);
+
+            asset fund_rate = asset_divide(commodity_ex, currency_ex);
+            
+            if (fund_rate == get_exchange_rate(fund_it->pool_id)) {
+                // TODO: emit fee participation token
+                
+                pools pool_markets(get_self(), get_self().value);
+                auto pool_it = pool_markets.find(fund_it->pool_id);
+                check(pool_it != pool_markets.end(), ERR_POOL_NOT_FOUND);
+
+                pool_markets.modify(pool_it, get_self(), [&](auto &row) {
+                    row.commodity_reserve += fund_it->commodity;
+                    row.currency_reserve += fund_it->currency;
+                });
+
+                funding_attempts.erase(fund_it);
+            }
+            return;
+        }
+
+        case PROTO_VERSION.value : {  
+            // general protocol parsing
+            asset min = asset_from_string(memo_tokens[2]);
+            name recipient = name(memo_tokens[3]);
+            check(is_account(recipient), ERR_RECIPIENT_NOT_FOUND);
+            
+            // get first element of path
+            string path_str = memo_tokens[1];
+            vector<string> jumps = split(path_str, " ");
+            check(jumps.size() > 0, ERR_EMPTY_PATH);
+
+            vector<string> conversion_data = split(jumps.front(), "/");
+            check(conversion_data.size() == 2, ERR_MEMO_PARSING);
+            check(name(conversion_data[0]) == get_self(), ERR_INCORRECT_CONVERTER);
+
+            // find pool
+            symbol_code A = quantity.symbol.code();
+            symbol_code B = symbol_code(conversion_data[1]);
+
+            pools pool_markets(get_self(), get_self().value);
+            auto sym_index = pool_markets.get_index<"symbols"_n>();
+            auto pool_it = sym_index.find(symbols_get_index(A, B).value);
+
+            if (pool_it == sym_index.end())
+                pool_it = sym_index.find(symbols_get_index(B, A).value);
+
+            check(pool_it != sym_index.end(), ERR_POOL_NOT_FOUND);
+
+            // check if pool has funds
+            check(pool_it->commodity_reserve.amount > 0, ERR_POOL_NOT_FUNDED);
+            check(pool_it->currency_reserve.amount > 0, ERR_POOL_NOT_FUNDED);
+
+            // make conversion
```

Comment:
After conversion, each converter should register a history entry in the correct market scope just like a normal deal for an order in the existing Telos (book) DEX:
https://github.com/guilledk/vapaee-smart-contracts/blob/telospooldex/include/vapaee/dex/modules/exchange.hpp#L515